src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.cs

```diff
@@ -316,12 +316,15 @@ private void VerbatimBlock()
                 Span.ChunkGenerator = SpanChunkGenerator.Null;
             }
 
-            if (!At(CSharpSymbolType.WhiteSpace) && !At(CSharpSymbolType.NewLine))
+            if (IsNested)
             {
-                PutCurrentBack();
+                CompleteBlock(insertMarkerIfNecessary: false);
+            }
+            else if (At(CSharpSymbolType.WhiteSpace) || At(CSharpSymbolType.NewLine))
```

Comment:
Not entirely true. What if we have:

```C#
@if (...)
{
}   <p>
```

In this case we'd want to render the whitespace.